### DISCUSSION TITLE
Renaming some of the C#'s elements to match C#'s naming conventions.

### DISCUSSION THREAD

**Discussion by TheVeryStarlk** [godotengine/godot-proposals#6262] [Engine Core]
Some namings used in the project like `MouseModeEnum` and `DegToRad` are not common in C#'s projects, instead it would be named `MouseMode` and `DegreesToRadians`.
This idea provides no real benefit other than being more close to C#'s conventions.

**Comment by Zireael07** [comment]
Too late to do it for 4.0 anyway iiuc

**Reply by TheVeryStarlk** [reply]
I thought it might be quite late, but I'm not sure if it actually is; Because Godot 3.5 had 7 RCs, so maybe we still have some time.

**Comment by raulsntos** [comment]
You are right that .NET naming conventions recommend avoiding abbreviations such as `Deg` and `Rad`. And while we try to match .NET naming conventions, we also try to avoid deviating too much from Godot's names because those names are used in the Core documentation and GDScript (which is used by the majority of content you can find out there). The C# bindings are also generated so adding exceptions is complicated and brittle.

The issue with `MouseMode` and other enums that end with `Enum` is that, often, Godot's API uses the same name for an enum type and a class member, to avoid the conflict the C# bindings generator uses a suffix.
